app/models/foreman_xen/xenserver.rb

```diff
@@ -201,19 +198,12 @@ def create_vm(args = {})
         raise 'you can select at most one template type'
       end
       begin
-        vm = nil
         logger.info "create_vm(): custom_template_name: #{custom_template_name}"
         logger.info "create_vm(): builtin_template_name: #{builtin_template_name}"
         vm = (custom_template_name != '') ? create_vm_from_custom(args) : create_vm_from_builtin(args)
         vm.set_attribute('name_description', 'Provisioned by Foreman')
-        cpus = args[:vcpus_max]
-        if vm.vcpus_max.to_i < cpus.to_i
-          vm.set_attribute('VCPUs_max', cpus)
```

Comment:
Possibly, but as in some other places `vcpus_max` is used like the simple CPU number setting in other compute resources I would prefer to scrap that differentiation here, especially as this behaviour went unnoticed since day 1 of this plugin being available.